pkg/apis/settings/install/install.go

```diff
@@ -22,17 +22,15 @@ import (
 	"k8s.io/apimachinery/pkg/apimachinery/announced"
 	"k8s.io/apimachinery/pkg/apimachinery/registered"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/apis/settings"
 	"k8s.io/kubernetes/pkg/apis/settings/v1alpha1"
 )
 
-func init() {
-	Install(api.GroupFactoryRegistry, api.Registry, api.Scheme)
-}
-
 // Install registers the API group and adds types to a scheme
 func Install(groupFactoryRegistry announced.APIGroupFactoryRegistry, registry *registered.APIRegistrationManager, scheme *runtime.Scheme) {
+	if _, ok := groupFactoryRegistry[settings.GroupName]; ok {
```

Comment:
Can't we move these checks into `Announce()` and `RegisterAndEnable` by making them idem-potent? These three lines are ugly.